modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -538,6 +538,45 @@ def reindex(self, axis, labels, **kwargs):
         return self.__constructor__(new_modin_frame)
 
     def reset_index(self, **kwargs):
+        if self._modin_frame._index_cache is None:
+
+            def _reset(df, *columns_idx, partition_idx):
+                _kw = dict(kwargs)
+                if len(columns_idx) > 1 and partition_idx == 0:
+                    idx = columns_idx[0].append(columns_idx[1:])
+                    cols = (
+                        pandas.DataFrame(index=df.index, columns=idx)
+                        .reset_index(**kwargs)
+                        .columns
+                    )
+                    len_new_cols = len(cols) - len(idx)
+
+                if partition_idx != 0:
+                    _kw["drop"] = True
+                res = df.reset_index(**_kw)
+                if len(columns_idx) > 1 and partition_idx == 0:
+                    res.columns = cols[: len(columns_idx[0]) + len_new_cols]
+                return res
+
+            return self.__constructor__(
+                self._modin_frame.broadcast_apply_full_axis(
+                    axis=0,
+                    func=_reset,
+                    other=None,
+                    enumerate_partitions=True,
+                    give_columns=True,
+                )
+            )
+
+        # Error checking for matching pandas. Pandas does not allow you to
+        # insert a dropped index into a DataFrame if these columns already
+        # exist.
+        if (
+            not kwargs["drop"]
+            and not self.has_multiindex()
+            and all(n in self.columns for n in ["level_0", "index"])
+        ):
+            raise ValueError("cannot insert level_0, already exists")
```

Comment:
Haven't we tested it yet?

Reply:
We tested it, but the check was not at the level of query compiler, but higher.

Reply:
Why doesn't codecov see we check this branch?

Reply:
I don't see this problem anymore.

Reply:
Why, I am still seeing.

```
modin/core/storage_formats/pandas/query_compiler.py#L582

Added line #L582 was not covered by tests
```

Reply:
This check does not work correctly. It does not take into account that the index may have a name other than `None`.

I will create a issue for this after we merge this pull request to include its new location in the issue description.

P.S. Apparently, it was the exception that was not tested before, or something changed, but before my pull request.

Reply:
https://github.com/modin-project/modin/issues/5635